Goal: Fix su syslogs to be less ambiguous.  (old:new instead of old-new
      because '-' can appear in usernames.)  Not clearer, mind you, but less
      ambiguous.
Fixes: #213592

Status wrt upstream: Applied upstream.

Index: shadow-4.0.3/src/su.c
===================================================================
--- shadow-4.0.3.orig/src/su.c	2005-05-29 12:11:25.560552000 +0200
+++ shadow-4.0.3/src/su.c	2005-05-29 12:12:22.910552000 +0200
@@ -228,7 +228,7 @@
 #ifdef USE_SYSLOG
 	if (getdef_bool ("SYSLOG_SU_ENAB"))
 		SYSLOG ((pwent.pw_uid ? LOG_INFO : LOG_NOTICE,
-			 "- %s %s-%s", tty,
+			 "- %s %s:%s", tty,
 			 oldname[0] ? oldname : "???",
 			 name[0] ? name : "???"));
 	closelog ();
@@ -678,7 +678,7 @@
 #endif
 #ifdef USE_SYSLOG
 	if (getdef_bool ("SYSLOG_SU_ENAB"))
-		SYSLOG ((LOG_INFO, "+ %s %s-%s", tty,
+		SYSLOG ((LOG_INFO, "+ %s %s:%s", tty,
 			 oldname[0] ? oldname : "???",
 			 name[0] ? name : "???"));
 #endif
